From b558f245d381bb67cc61989c4cd82ef83f85de0e Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 29 Aug 2016 15:01:50 +0200
Subject: [PATCH] net: phy: ti: enable extended read / write for driver

this make extended rx/wx usable from mdio command

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
(cherry picked from commit 4d5c2bce1fd766dee172c141399a0d157ed5718b)
---
 drivers/net/phy/ti.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/net/phy/ti.c b/drivers/net/phy/ti.c
index 3add1e2566..03cfee23ab 100644
--- a/drivers/net/phy/ti.c
+++ b/drivers/net/phy/ti.c
@@ -164,6 +164,23 @@ void phy_write_mmd_indirect(struct phy_device *phydev, int prtad,
 	phy_write(phydev, addr, MII_MMD_DATA, data);
 }
 
+static int dp83867_phy_extread(struct phy_device *phydev,
+			       int addr, int devad, int reg)
+{
+	if (devad != DP83867_DEVADDR)
+		return -1;
+	return phy_read_mmd_indirect(phydev, reg, devad, addr);
+};
+
+static int dp83867_phy_extwrite(struct phy_device *phydev, int addr,
+				int devad, int reg, u16 val)
+{
+	if (devad != DP83867_DEVADDR)
+		return -1;
+	phy_write_mmd_indirect(phydev, reg, devad, addr, (u32)val);
+	return 0;
+};
+
 #if defined(CONFIG_DM_ETH)
 /**
  * dp83867_data_init - Convenience function for setting PHY specific data
@@ -329,6 +346,8 @@ static struct phy_driver DP83867_driver = {
 	.config = &dp83867_config,
 	.startup = &genphy_startup,
 	.shutdown = &genphy_shutdown,
+	.readext = dp83867_phy_extread,
+	.writeext = dp83867_phy_extwrite,
 };
 
 int phy_ti_init(void)
